lib/ansible/plugins/test/core.py

```diff
@@ -142,14 +142,17 @@ def regex(value='', pattern='', ignorecase=False, multiline=False, match_type='s
         This is likely only useful for `search` and `match` which already
         have their own filters.
     """
+    valid_match_types = ('search', 'match', 'fullmatch')
+    if match_type not in valid_match_types:
+        raise errors.AnsibleTemplatePluginError(f"Invalid match_type specified. Expected one of: {valid_match_types}.", obj=match_type)
```

Comment:
This change will yield a better error message:

```suggestion
        raise errors.AnsibleTemplatePluginError(f"Invalid match_type specified. Expected one of: {', '.join(valid_match_types)}.", obj=match_type)
```

`Expected one of: search, match, fullmatch.`

Compared to:

`Expected one of: ('search', 'match', 'fullmatch').`